packages/pointer_interceptor/pointer_interceptor_platform_interface/lib/src/pointer_interceptor_platform.dart

```diff
@@ -0,0 +1,45 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'default_pointer_interceptor.dart';
+
+/// Platform-specific implementations should set this with their own
+/// platform-specific class that extends [PointerInterceptorPlatform] when
+/// they register themselves.
+abstract class PointerInterceptorPlatform extends PlatformInterface {
+  /// Constructs a PointerInterceptorPlatform.
+  PointerInterceptorPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static PointerInterceptorPlatform _instance = DefaultPointerInterceptor();
+
+  static set instance(PointerInterceptorPlatform? instance) {
+    if (instance == null) {
+      throw AssertionError(
+          'Platform interfaces can only be set to a non-null instance');
+    }
+
+    PlatformInterface.verify(instance, _token);
+    _instance = instance;
+  }
+
+  /// The default instance of [PointerInterceptorPlatform] to use.
+  ///
+  /// Defaults to [DefaultPointerInterceptor], which does not do anything
+  static PointerInterceptorPlatform get instance => _instance;
+
+  /// Platform-specific implementations should override this function their own
+  /// implementation of a pointer interceptor widget.
+  Widget buildWidget({
```

Comment:
We should have a Dart unit test in this PR that calls this and asserts that it throws UnimplementedError; see [here](https://github.com/flutter/packages/blob/fcb1272be272348f32456e5f3e5346b071f8f7c1/packages/webview_flutter/webview_flutter_platform_interface/test/webview_platform_test.dart#L106-L116) for an example. (This is partly because we actually care that the default is to throw, and partially to make sure we have unit tests set up so future changes can more easily add tests.)

Reply:
By default it returns it's child because thats what the no-op pointer interceptor does, which is set as the default instance

Reply:
Well, we should test that too in a Dart unit test of the default implementation class. But this code you would test in by making a test subclass with no overrides (e.g., `class UnimplementedPointerInterceptorPlatform extends PointerInterceptorPlatform {}`, and then testing that.

Alternately, since this is a new class, you could make it abstract and not provide an implementation of this method at all. The `Unimplemented` pattern is something we do a lot because any method added after publishing must have a default implementation to avoid breaking compilation of existing code, but you don't have that problem here.

Reply:
Ohhh, my bad, I see what you mean about the default class now!